lib/linguist/heuristics.yml

```diff
@@ -453,16 +453,15 @@ disambiguations:
     pattern: '(?i:^\\i\b|AS \$\$|LANGUAGE ''?plpgsql''?|BEGIN( WORK )?;)'
   # IBM db2
   - language: SQLPL
-    pattern: "(?i:alter module|MODE DB2SQL)"
+    pattern: '(?i:alter module|MODE DB2SQL|\bSYSCAT\.\b|\bSYSPROC\.\b|ASSOCIATE RESULT SET|\bEND!\s*$)'
   # Oracle
   - language: PLSQL
     pattern: '(?i:\$\$PLSQL_|XMLTYPE|systimestamp|\.nextval|connect by|AUTHID (DEFINER|CURRENT_USER)|constructor\W+function)'
   # T-SQL
   - language: TSQL
-    pattern: '(?i:^\s*GO\b|BEGIN( TRY| CATCH)|OUTPUT INSERTED|DECLARE\s+@)'
-  # Generic SQL
+    pattern: '(?i:^\s*GO\b|BEGIN( TRY| CATCH)|OUTPUT INSERTED|DECLARE\s+@|\[dbo\])'
```

Comment:
Here's something a little clearer:

~~~regex
(?i:^\s*GO\b|BEGIN +(TRY|CATCH)|OUTPUT +INSERTED|DECLARE\s+@|\[dbo\])
~~~

Note that if vertical whitespace is permitted where ` +`is used above, then the pattern should instead be:

~~~regex
(?i:^\s*GO\b|BEGIN\s+(TRY|CATCH)|OUTPUT\s+INSERTED|DECLARE\s+@|\[dbo\])
~~~

Reply:
Changed everything to `\s+`, indeed, all kinds of spaces are allowed.